server/controllers/uploadController.js

```diff
@@ -0,0 +1,36 @@
+const aws = require("aws-sdk");
+const multer = require("multer");
+const multers3 = require("multer-s3");
+const Profile = require("../models/ProfileModel");
+
+const s3 = new aws.S3({
+  accessKeyId: process.env.S3_ACCESS_KEY,
+  secretAccessKey: process.env.S3_SECRET_KEY,
+  region: process.env.S3_BUCKET_REGION,
+});
+
+exports.setUpload = (req, res, next) => {
+  console.log(req.file);
+  const upload = multer({
+    storage: multers3({
+      s3,
+      bucket: "breadsticks",
+      metadata: function (req, file, cb) {
+        cb(null, { fieldName: file.fieldName });
+      },
+      key: function (req, file, cb) {
+        cb(null, `image-${Date.now()}.jpeg`);
+      },
+    }),
+  });
+  const multipleUpload = upload.array("image-upload", 12);
+  multipleUpload(req, re, async (err) => {
```

Comment:
Don't understand what you're trying to do here.

Reply:
the multer object saved in the variable upload then add method array() to allow upload multiple files.
the string "imge-upload" is the name of the key you choose to put in the value of the file
12 is the number of files allowed to be uploaded.
then use the other function multipleUpload  to update the database profile schema , by adding the names of the files to it.